interpreter/Makefile

```diff
@@ -142,67 +55,47 @@ TESTDIR =	../test/core
 TESTFILES =	$(shell cd $(TESTDIR); ls *.wast; ls [a-z]*/*.wast)
 TESTS =		$(TESTFILES:%.wast=%)
 
-.PHONY:		test debugtest partest dune-test
+.PHONY:		test partest dune-test
 
 test:		$(OPT) smallint
-		$(TESTDIR)/run.py --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',)
-		./smallint.native
-debugtest:	$(UNOPT) smallint
-		$(TESTDIR)/run.py --wasm `pwd`/$(UNOPT) $(if $(JS),--js '$(JS)',)
-		./smallint.native
+		$(TESTDIR)/run.py --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',)
+		dune exec ./smallint.exe
 
 test/%:		$(OPT)
-		$(TESTDIR)/run.py --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
-debugtest/%:	$(UNOPT)
-		$(TESTDIR)/run.py --wasm `pwd`/$(UNOPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
+		$(TESTDIR)/run.py --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast
 
 run/%:		$(OPT)
 		./$(OPT) $(TESTDIR)/$*.wast
-debug/%:	$(UNOPT)
-		./$(UNOPT) $(TESTDIR)/$*.wast
 
-partest: 	$(TESTS:%=quiettest/%)
+partest:	$(TESTS:%=quiettest/%)
 		@echo All tests passed.
 
 quiettest/%:	$(OPT)
 		@ ( \
-		  $(TESTDIR)/run.py 2>$(@F).out --wasm `pwd`/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast && \
+		  $(TESTDIR)/run.py 2>$(@F).out --wasm `pwd`/$(BUILDDIR)/$(OPT) $(if $(JS),--js '$(JS)',) $(TESTDIR)/$*.wast && \
 		  rm $(@F).out \
 		) || \
 		cat $(@F).out || rm $(@F).out || exit 1
 
 smallinttest:	smallint
-		@./smallint.native
+	dune exec ./smallint.exe
 
 dunetest:
 	dune test
 
+install:
+	dune build -p $(NAME) @install
+	dune install
 
 # Miscellaneous targets
 
 .PHONY:		clean
 
-$(ZIP):		$(WINMAKE)
-		git archive --format=zip --prefix=$(NAME)/ -o $@ HEAD
+$(ZIP):
+	git archive --format=zip --prefix=$(NAME)/ -o $@ HEAD
 
 clean:
-		rm -rf _build/jslib $(LIB).mlpack _tags $(JSLIB).js
-		$(OCB) -clean
-
-
-# Opam support
-
-.PHONY:		check install uninstall
-
-check:
-		# Check that we can find all relevant libraries
-		# when using ocamlfind
-		ocamlfind query $(LIBS)
-
-install:	_build/$(LIB).cmx _build/$(LIB).cmo
```

Comment:
This target (as well as opt and unopt) is currently used by the `opam` package file. I'm not sure what that needs to be replaced with.

As part of a doing a new 2.0.1 package this morning, I checked in a template for the opam file under meta/opam. Can you merge main and adapt that file to the changes here?

Reply:
They won't be needed anymore once we use dune.

I updated the `dune-project` to add the new metadata from `meta/opam`.